src/core_plugins/i18n/server/__tests__/i18n.js

```diff
@@ -0,0 +1,278 @@
+        checkTranslations(locale, {});
+        return;
+      });
+
+    });
+
+    describe('getRegisteredTranslationLocales', function () {
+
+      it('Translation locales are registered', function () {
+        const expectedLocales = ['en', 'de'];
+        checkRegisteredLocales(expectedLocales);
+        return;
+      });
+
+    });
+
+    describe('getTranslationsForLocales', function () {
+
+      it('no translations as no locale registered which matches, then get translations for default locale' , function () {
+        const locales = [
+          { code: 'es', region: 'ES', quality: 1.0 },
+          { code: 'fr', region: undefined, quality: 0.8 }
+        ];
+        const expectedTranslations = {
+          'test_plugin_1-NO_SSL': 'Dont run the dev server using HTTPS',
+          'test_plugin_1-DEV': 'Run the server with development mode defaults',
+          'test_plugin_1-NO_RUN_SERVER': 'Dont run the dev server',
+          'test_plugin_1-HOME': 'Run along home now!'
+        };
+        const server = {
+          config: {
+            get: function () {
+              return 'en';
+            }
+          }
+        };
+        checkTranslationsForLocales(getLanguageTags(locales), {});
+        checkTranslationsForDefaultLocale(server, expectedTranslations);
+        return;
+      });
+
+      it('de translations should be chosen' , function () {
+        const locales = [
+          { code: 'es', region: 'ES', quality: 1.0 },
+          { code: 'de', region: undefined, quality: 0.8 }
+        ];
+        const expectedTranslationJson = {
+          'test_plugin_1-NO_SSL': 'Dont run the DE dev server using HTTPS',
+          'test_plugin_1-DEV': 'Run the DE server with development mode defaults',
+        };
+        checkTranslationsForLocales(getLanguageTags(locales), expectedTranslationJson);
+        return;
+      });
+
+      it('No translations regsitered for locales, so empty translations returned' , function () {
+        const locales = [
+          { code: 'es', region: 'ES', quality: 1.0 },
+          { code: 'fr', region: undefined, quality: 0.8 }
+        ];
+        const expectedTranslationJson = {
+          'test_plugin_1-NO_SSL': 'Dont run the dev server using HTTPS',
+          'test_plugin_1-DEV': 'Run the server with development mode defaults',
+          'test_plugin_1-NO_RUN_SERVER': 'Dont run the dev server',
+          'test_plugin_1-HOME': 'Run along home now!'
+        };
+        checkTranslationsForLocales(getLanguageTags(locales), {});
+        return;
+      });
+
+    });
+
+  });
+
+  describe('multiple plugins', function () {
+
+    before('registerTranslations - multiple plugin', function () {
+      let pluginName = 'test_plugin_1';
+      let pluginTranslationPath = path.join(FIXTURES, 'translations', pluginName);
+      let translationFiles = [
+        path.join(pluginTranslationPath, 'de.json'),
+        path.join(pluginTranslationPath, 'en.json')
+      ];
```

Comment:
Why not just do all the translation files for both plugins at once here? It would be easier to understand:

```
const translationDir = path.join(FIXTURES, 'translations');
const translationFiles = [
  path.join(translationDir, 'test_plugin_1', 'de.json'),
  path.join(translationDir, 'test_plugin_1', 'en.json'),
  path.join(translationDir, 'test_plugin_2', 'de.json'),
  path.join(translationDir, 'test_plugin_2', 'en.json'),
];
translationFiles.forEach(i18n.registerTranslations);
```